@@ -, +, @@ 
 sys/v4l2/gstv4l2object.c | 7 +++++++
 1 file changed, 7 insertions(+)
--- a/sys/v4l2/gstv4l2object.c	
+++ a/sys/v4l2/gstv4l2object.c	
@@ -2954,6 +2954,13 @@ gst_v4l2_object_extrapolate_info (GstV4l2Object * v4l2object,
         "stride %d, offset %" G_GSIZE_FORMAT, i, stride, info->stride[i],
         info->offset[i]);
   }
+
+  /* Update the image size according the amount of data we are going to
+   * read/write. This workaround bugs in driver where the sizeimage provided
+   * by TRY/S_FMT represent the buffer length (maximum size) rather then the expected
+   * bytesused (buffer size). */
+  if (offs < info->size)
+    info->size = offs;
 }
 
 static void
